### DISCUSSION TITLE
需要有人适配platfromIO

### DISCUSSION THREAD

**Discussion by qnxsgwy** [Air-duino/Arduino-AirMCU#80] [General]
Arduino支持已经挺完善了，platfromIO无非就是多写个配置文件的事，希望尽快支持platfromIO

**Comment by zgjsntsm** [comment]
同求，PIO的体验更好。

[a comment by an automated account is not shown]

**Comment by santook** [comment]
试试Visual studio+Visual Micro吧，PIO下载太拉跨

**Comment by zgjsntsm** [comment]
> 

太重了，且只支持arduino

**Comment by chenxuuu** [comment]
都开源的，你们可以自己适配，毕竟新建issue的作者也说了，就一个配置文件的事



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年7月11日(周二) 下午5:47
收件人: ***@***.***&gt;;
抄送: ***@***.***&gt;;
主题: Re: [Air-duino/Arduino-AirMCU] 希望尽快支持platfromIO，Arduino IDE用着太拉跨。 (Issue #4)




 
试试Visual studio+Visual Micro吧，PIO下载太拉跨

**Comment by qnxsgwy** [comment]
> 都开源的，你们可以自己适配，毕竟新建issue的作者也说了，就一个配置文件的事
> […](#)
> ---原始邮件--- 发件人: ***@***.***&gt; 发送时间: 2023年7月11日(周二) 下午5:47 收件人: ***@***.***&gt;; 抄送: ***@***.***&gt;; 主题: Re: [Air-duino/Arduino-AirMCU] 希望尽快支持platfromIO，Arduino IDE用着太拉跨。 (Issue #4) 试试Visual studio+Visual Micro吧，PIO下载太拉跨 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&gt;

但是想收录到Pio官方的支持列表，开箱即用。Pio的Platform，Frameworks，Boards一系列配置文件都需要给官方Pull requests，为了长期问题维护，还是你们官方提Pull更合适吧。

**Comment by chenxuuu** [comment]
是的，提交当然是官方来搞，但是配置文件大家可以贡献，毕竟开发人员里没有人用platform io



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2023年7月13日(周四) 凌晨0:56
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [Air-duino/Arduino-AirMCU] 希望尽快支持platfromIO，Arduino IDE用着太拉跨。 (Issue #4)




   
但是想收录到Pio官方的支持列表，开箱即用。Pio的Platform，Frameworks，Boards一系列配置文件都需要给官方Pull requests，为了长期问题维护，还是你们官方提Pull更合适吧。

**Comment by HalfSweet** [comment]
> > 都开源的，你们可以自己适配，毕竟新建issue的作者也说了，就一个配置文件的事
> > […  ……](#)
> > ---原始邮件--- 发件人: _**@**_._**> 发送时间: 2023年7月11日(周二) 下午5:47 收件人: _**@**_.**_>; 抄送: _**@**_._**>; 主题: Re: [Air-duino/Arduino-AirMCU] 希望尽快支持platfromIO，Arduino IDE用着太拉跨。 (Issue #4) 试试Visual studio+Visual Micro吧，PIO下载太拉跨 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: _**@**_.**_> ---原始邮件--- 发件人：@.> 发送时间：2023年7月11日(周二)下午5:47 早餐：@.>;抄送：@.>;主题: Re: [Air-duino/Arduino-AirMCU]希望加速支持platfromIO，Arduino IDE用着着太拉跨。 (Issue #4) 尝试Visual studio+Visual Micro吧，PIO下载太拉跨 — 回复本邮件直接、在 GitHub 上查看或取消订阅。您收到此消息是因为您订阅了该主题。消息 ID：@。>
> 
> 但是想收录到Pio官方的支持列表，开箱即用。Pio的Platform，Frameworks，Boards一系列配置文件都需要给官方Pull requests，为了长期问题维护，还是你们官方提Pull更合适吧。

PIO的文档太少了，不太好搞，如果有大佬可以做一套完整的描述文件/CI自动上传包的话可以联系我<HalfSweet@HalfSweet.cn>或者直接提交Pr到仓库，我们来进行上传到官方的包管理系统

目前的PIO的仓库独立在<https://github.com/Air-duino/platform-airmcu>下

**Comment by jyzhkj** [comment]
同求呀，PIO好用得多了。

给个例子吧, 肯定有帮助 ：
国内仿avr的芯片lgt 的arduino
arduino：https://github.com/dbuezas/lgt8fx
PIO: https://github.com/darkautism/pio-lgt8fx

**Comment by BKTM-2020** [comment]
支持官方移植到platfromIO，这个平台比arduinoide 好很多 现在用的人是越来越多

**Comment by MaShuilong** [comment]
适配PIO的时候记得拍一下我

**Comment by CodeWenjiu** [comment]
aaaaa什么时候适配platformio

**Comment by HalfSweet** [comment]
根据pio的官方定价策略：https://registry.platformio.org/pricing

想要发布platform以及tools包，至少得需要Pro会员，但是每月的续费高达10美元，这对于一个社区项目来说成本似乎有些高了。

因此这个计划将长期搁置，直到我有足够的小钱钱能付得起会员，或者有其他赞助来源吧QwQ

~~我就说为什么之前我ci的包死活publish不上去，原来是要加钱啊~~
